glean-core/ios/Glean/Net/HttpPingUploader.swift

```diff
@@ -122,4 +127,91 @@ public class HttpPingUploader {
         dateFormatter.setLocalizedDateFormatFromTemplate("EEE, dd MMM yyyy HH:mm:ss z")
         return dateFormatter.string(from: date)
     }
+
+    /// This function deserializes and processes all of the serialized ping files.
+    ///
+    /// This function will ignore files that don't match the UUID regex and just delete them to
+    /// prevent files from polluting the ping storage directory.
+    func process() {
+        let pingDirectory = HttpPingUploader.getOrCreatePingDirectory()
+
+        do {
+            let storageDirectory = try FileManager.default.contentsOfDirectory(
+                atPath: pingDirectory.absoluteString
+            )
+
+            for file in storageDirectory {
+                if file.lastPathComponent.matches(Constants.filePattern) {
+                    NSLog("\(Constants.logTag) : Processing ping: \(file)")
+                    processFile(pingDirectory.appendingPathComponent(file)) { success, error in
+                        if !success {
+                            NSLog(
+                                "\(Constants.logTag) : Error processing ping file: \(file) - \(error.debugDescription)"
+                            )
+                        } else {
+                            do {
+                                try FileManager.default.removeItem(
+                                    atPath: pingDirectory.appendingPathComponent(file).absoluteString
+                                )
+                            } catch {
+                                print("\(Constants.logTag) : Error deleting ping file: \(file)")
+                            }
+                        }
+                    }
+                } else {
+                    // Delete files that don't match the UUID filePattern regex
+                    NSLog("\(Constants.logTag) : Pattern mismatch. Deleting \(file)")
+                    try FileManager.default.removeItem(
```

Comment:
Will `try` just explode if stuff fails?

Reply:
no the catch below handles all of the try's for the FileManager stuff in this do/catch block.